### ISSUE TITLE
Cannot run at the same time as pimd

### ISSUE THREAD

**Issue by GrumpyOldTroll** [troglobit/pim6sd#28]
It looks not possible to run this concurrently with pimd to support a dual-stack multicast routing?

If I try I get an error message and a stop:

~~~
cannot enable multicast routing in kernel: Address already in use
~~~

**Comment by GrumpyOldTroll** [comment]
A quick glance suggests that it's intentional in the kernel to deny this if using the same table:
- https://github.com/torvalds/linux/blob/9ff9b0d392ea08090cd1780fb196f36dbb586529/net/ipv4/ipmr.c#L1386
- https://github.com/torvalds/linux/blob/fcadab740480e0e0e9fa9bd272acd409884d431a/net/ipv6/ip6mr.c#L1554

Is there maybe a way to use different mrt tables for default IPv4 and default IPv6?

**Comment by troglobit** [comment]
Yup, they share the same multicast routing socket by default.  I have it on my backlog of things to port the multiple-table support from pimd to pim6sd, that would indeed fix this limitation.

**Comment by ne-vlezay80** [comment]
Please paste kernel version

**Comment by troglobit** [comment]
@ne-vlezay80 kernel version doesn't help when pim6sd doesn't support multicast policy routing yet.

What may help is to use the pimd functionality and let pim6sd use the "main" table, as long as there is no overlap between interfaces.  Should have mentioned that possible workaround previously.

**Comment by ne-vlezay80** [comment]
> What may help is to use the pimd functionality and let pim6sd use the "main" table, as long as there is no overlap between interfaces. Should have mentioned that possible workaround previously.

And How

**Comment by troglobit** [comment]
@ne-vlezay80 Well I haven't tried it, that's why I said "possible workaround".  It's entirely likely to not work at all, but you can play around with it yourself, the pimd docs detail the table option.
